pkg/rancher-desktop/config/__tests__/settings.spec.ts

```diff
@@ -72,10 +249,255 @@ describe('updateFromCommandLine', () => {
+        test('all fields are unlocked', () => {
+          settings.load();
+          verifyAllFieldsAreUnlocked(settings.getLockedSettings());
+        });
+      });
+
+      describe('when there is only a user profile with locked imageList', () => {
+        beforeEach(() => {
+          mock = jest.spyOn(fs, 'readFileSync')
+            .mockImplementation(createMocker(0, 2));
+        });
+        afterEach(() => {
+          mock.mockRestore();
+        });
+        test('all fields are locked', () => {
+          settings.load();
+          verifyAllFieldsAreLocked(settings.getLockedSettings());
+        });
+      });
+
+      describe('when there is only a system profile with unlocked imageList', () => {
+        beforeEach(() => {
+          mock = jest.spyOn(fs, 'readFileSync')
+            .mockImplementation(createMocker(1, 0));
+        });
+        afterEach(() => {
+          mock.mockRestore();
+        });
+        test('all fields are unlocked', () => {
+          settings.load();
+          verifyAllFieldsAreUnlocked(settings.getLockedSettings());
+        });
+      });
+
+      describe('when both profiles exit, both with unlocked imageList', () => {
+        beforeEach(() => {
+          mock = jest.spyOn(fs, 'readFileSync')
+            .mockImplementation(createMocker(1, 1));
+        });
+        afterEach(() => {
+          mock.mockRestore();
+        });
+        test('all fields are locked', () => {
+          settings.load();
+          verifyAllFieldsAreUnlocked(settings.getLockedSettings());
+        });
+      });
+
+      describe('when the system profile is unlocked and the user profile is locked', () => {
+        beforeEach(() => {
+          mock = jest.spyOn(fs, 'readFileSync')
+            .mockImplementation(createMocker(1, 2));
+        });
+        afterEach(() => {
+          mock.mockRestore();
+        });
+        test('all fields are locked', () => {
+          settings.load();
+          verifyAllFieldsAreUnlocked(settings.getLockedSettings());
+        });
+
+        describe('when there is only a system profile with locked imageList', () => {
+          beforeEach(() => {
+            mock = jest.spyOn(fs, 'readFileSync')
+              .mockImplementation(createMocker(2, 0));
+          });
+          afterEach(() => {
+            mock.mockRestore();
+          });
+          test('all fields are locked', () => {
+            settings.load();
+            verifyAllFieldsAreLocked(settings.getLockedSettings());
+          });
+        });
+
+        describe('when both profiles exit, system locked, user unlocked', () => {
+          beforeEach(() => {
+            mock = jest.spyOn(fs, 'readFileSync')
+              .mockImplementation(createMocker(2, 1));
+          });
+          afterEach(() => {
+            mock.mockRestore();
+          });
+          test('all fields are locked', () => {
+            settings.load();
+            verifyAllFieldsAreLocked(settings.getLockedSettings());
```

Comment:
Should we be testing to make sure the locked settings win?

Reply:
I addressed this later... I believe I am